schemas/2016-04-01/Microsoft.Cache.json

```diff
@@ -127,16 +237,201 @@
               "pattern": "^\\d+\\.\\d+\\.\\d+\\.\\d+$"
             },
             {
-              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+              "$ref": "https://schema.management.azure.com/schemas/common/definitions.json#/definitions/expression"
             }
           ],
-          "description": "Required when deploying a redis cache inside an existing Azure Virtual Network."
+          "description": "Static IP address. Required when deploying a Redis cache inside an existing Azure Virtual Network."
+        },
+        "sku": {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/Sku"
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/common/definitions.json#/definitions/expression"
+            }
+          ],
+          "description": "SKU parameters supplied to the create Redis operation."
         }
       },
       "required": [
         "sku"
       ],
-      "description": "Properties supplied to CreateOrUpdate redis operation."
+      "description": "Properties supplied to Create Redis operation."
+    },
+    "RedisFirewallRuleProperties": {
+      "type": "object",
+      "properties": {
+        "startIP": {
```

Comment:
startIP and endIP - another place we should probably allow for template expressions to be valid

Reply:
They're already valid here (since they're strings). If you want to make the IP validation logic stricter, you can do so by making a change here: https://github.com/Azure/azure-rest-api-specs/blob/1001207991f2c9eb955cc4ce8ade14a72f2d20ee/specification/redis/resource-manager/Microsoft.Cache/stable/2016-04-01/redis.json#L1180-L1188